### PR TITLE
A first attempt at storing generated po files in sorted order. 

### PR DESCRIPTION
**Pull request by monkeyiq**
This should hopefully minimize the number of changes that git sees when storing these files in a repo

Note that some of the header metadata is not cloned at the moment. If this ordering fixes the issue that was causing git chatter then the metadata clone can be added before merge.

Tested using
```
cd ./bin
./translations translations:update:translatable
vi ../locales/en/LC_MESSAGES/messages.po
```

### CODE REVIEW

File: src/SimpleSAML/Command/UpdateTranslatableStringsCommand.php [hunk 1]
```diff
@@ -31,6 +31,22 @@
 use function ksort;
 use function sprintf;
 
+
+function cloneIteratorToTranslations($ret, $iterator)
```
Comment (tvdijen) [inline src/SimpleSAML/Command/UpdateTranslatableStringsCommand.php:35]: This should probably be private method inside the class.. The parameters should be type-hinted.

**Comment by monkeyiq** [comment]
I guess the big question is whether having the keys in the generated po files in sorted order is the main issue that was causing the git chatter or not. Hopefully I understood the issue enough.

File: src/SimpleSAML/Command/UpdateTranslatableStringsCommand.php (hunk 1 above)
Comment (monkeyiq) [reply src/SimpleSAML/Command/UpdateTranslatableStringsCommand.php:34]: Agreed. I did mention that this was a quick and dirty PoC :)

**Comment by tvdijen** [comment]
> I guess the big question is whether having the keys in the generated po files in sorted order is the main issue that was causing the git chatter or not. Hopefully I understood the issue enough.

I think you did! This PR won't hurt anyone if it doesn't make it's promise, so feel free to fix the review comments and merge!

**Comment by monkeyiq** [comment]
I should probably also look at cloning the metadata too before the merge

File: src/SimpleSAML/Command/UpdateTranslatableStringsCommand.php (hunk 1 above)
Comment (tvdijen) [reply src/SimpleSAML/Command/UpdateTranslatableStringsCommand.php:34]: You did ;)  We're strict like this! :goberserk:

**Comment by tvdijen** [comment]
What metadata are we talking about? I think we keep comments, headers and references.. Is there more we would want/need? I feel like the translations-files should not become a source of documentation..

**Comment by tvdijen** [comment]
> I think there are some in the attributes.po
> [..]
> Though it seems that file is not processed by `translations translations:update:translatable`.

Correct. The `attributes.po` file is not processed at all... It's a leftover that still needs work. I believe there is also an open issue to fix the `bin/translateAttributes.sh` script

File: src/SimpleSAML/Command/UpdateTranslatableStringsCommand.php (hunk 1 above)
Comment (monkeyiq) [reply src/SimpleSAML/Command/UpdateTranslatableStringsCommand.php]: Updated, https://github.com/simplesamlphp/simplesamlphp/pull/1961/commits/9a09e5e89415fa8d56525e0b1c408f9582956c59

**Comment by monkeyiq** [comment]
So this should be ok now, sans the awaiting workflow. I seem to not have write access at the moment so can not self merge.

**Comment by monkeyiq** [comment]
translateAttributes.sh might be something I can take a look at after this to see if there is something I can do there too.

**Comment by tvdijen** [comment]
I've enabled the workflow and sent you an invitation to the developers-team which should grant you access to all repositories.
The linter-workflow is known to fail, so you can forget about that one.
Since this file is not covered at all by unit tests, I'd suggest you fix the quality-workflow and just merge..

**Comment by monkeyiq** [comment]
I'll double check this tomorrow and merge.

**Comment by monkeyiq** [comment]
I reverified that locales/en/LC_MESSAGES/messages.po and the original messages.po are the same data when both are sorted.

**Pull request merged**